### DISCUSSION TITLE
Fetching single playbook form URL instead of entire repo

### DISCUSSION THREAD

**Discussion by knuurr** [semaphoreui/semaphore#1471] [Q&A] [answered]
Hi all,

I use Semaphore to run some maintenance tasks via localhost. I've ensured everything is properly configured including definition of localhost in my inventory file.

I'm using small, self-contained playbooks for each task I have. They do not need any external files, they do not reference any templates and are completely standalone.

After reading docs and searching for examples, it is not obvious for me. Is it possible that instead of fetching the same github repo everytime, where in this repo I keep other things fetch just that single particular playbook ?

Github provides direct links to each file as raw but I wonder if Semaphore allows to use it.

Of course this would mean that I would need to add multiple entires for each playbook I maintain but for my usage that's ok.

Fetching the same github repo is a waste of badwidth and resources, also I see that I must've been hittingsome timeouts during tests.

**Comment by fiftin** [comment, marked as answer]
Hi,

no, it is not possible. Git fetches whole repo only.

**Reply by knuurr** [reply]
Ok, thanks for reply. May we have it as feature request then? To be able to run standalone self-contained playbook from defined network endpoint? 
I mean under the hood, it's probably as complicated as wget file content into tmp file and run it.

**Comment by ansibleguy** [comment]
Just to mention it:
As a workaround you could fetch these files externally (and periodically update them..) and then use the repository-type 'path' to target the directory containing all your playbooks.

A simple cron + bash script could do the job.
Of course that is not a 'clean' way of doing it..

**Reply by knuurr** [reply]
I'm back after break from home labs. I agree, not clean. But from what I see nothing changed regarding this. It's hacky, but for my needs it may be sufficient.
